### PR TITLE
Format image url for social meta tags correctly and add analytics.

### PR DESCRIPTION
**Pull request by welteki**
Signed-off-by: Han Verstraete (OpenFaaS Ltd) <han@openfaas.com>

## Description

- Fix an issue where the URL for images int the `meta` tags was not formatted correctly.
- Add analytics.

## Motivation and Context

Image previews for post shared on social media where not displayed.

## How Has This Been Tested?

- Verified meta tags use a valid url:

```html
<meta name="twitter:image:src" content="https://actuated-blog.exit.welteki.dev/images/2023-native-arm64-for-oss/in-progress-dashboard.png">
<meta property="og:image" content="https://actuated-blog.exit.welteki.dev/images/2023-native-arm64-for-oss/in-progress-dashboard.png">
```

- Verified the analytics script is included on each page.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pages/_document.js [hunk 1]
```diff
@@ -20,6 +20,18 @@ export default function Document() {
         user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/ */}
         <link rel="manifest" href="/manifest.json" />
 
+        {/* Google tag (gtag.js) */}
```
Comment (alexellis) [inline pages/_document.js:23]: Does this add GA to the whole site, every page?

File: pages/blog/[id].js [hunk 2]
```diff
@@ -6,6 +6,8 @@ import Date from '../../components/date';
 const siteURL = process.env.PUBLIC_URL
 
 export default function Post({ post }) {
+  let imageURL = new URL(post.image, siteURL).toString();
```
Comment (alexellis) [inline pages/blog/[id].js:9]: This should be guarded with an if to prevent a null reference when not set.

File: pages/_document.js (hunk 1 above)
Comment (welteki) [reply pages/_document.js:23]: Yes this adds it to every page.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**